### ISSUE TITLE
Objections against clean-up refactorings?

### ISSUE THREAD

**Issue by Bananeweizen** [MoreUnit/MoreUnit-Eclipse#222]
During one of my last PRs I noticed that the sources would benefit from some mass cleanups. I imagine things like
* removing all author tags (that information is available via git nowadays)
* enabling some more compiler checks like enforcing the use of Override annotations, which allow to better detect dead code or broken class hierarchies
* conversion towards more recent Java features: instanceof with following type casts can be converted to "instanceof Type newVariable" nowadays, making everything more readable.
* improving the formatter to not join wrapped lines, such that Stream based operations can be wrapped over multiple lines without comments in between (everything else would remain as is)
* ...

Would anyone object such cleanups that don't fix bugs or introduce new features? I'm asking mainly because some projects object such changes "because it has bad impact on git blame", and I'd like to avoid wasted work. I really appreciate that so many PRs have been merged up to now. :)

**Comment by RoiSoleil** [comment]
That's ok for me.

**Comment by dstango** [comment]
Regarding formatting: I guess this will have the biggest impact on many files, potentially causing merge conflicts with other people working on features, and would thus propose to do this in a separate PR ASAP.

I'd also propose to have formatting settings closer to current default Eclipse Formatter settings, or e.g. google format.

One more idea: the maven build could include the spotless plugin that checks (or even corrects) the formatting: https://github.com/diffplug/spotless/tree/main/plugin-maven -- we started using it in some projects a while ago, and it's working fine.

**Comment by Bananeweizen** [comment]
The only change I intend to do with the formatter is to not join wrapped lines. That will have _zero_ effects on existing code, and will only affect new code, where stream.a().b().c() is split onto multiple lines (right now the formatter would make one line out of it again).

Regarding your suggestion: To me it doesn't make much sense to check or modify the formatting in Maven, when that can all be enforced via the formatter settings and clean on save actions in Eclipse itself. It's actually quite hard to have the settings of Eclipse and additional tools in sync (be it spotless, checkstyle, Sonarqube...), therefore it is definitely better to have them only once, fully automated, and as early as possible in the pipeline. I'm one of the main developers of eclipse-checkstyle, and I never use it for formatting checks in my work or spare time projects, but rather to avoid unwanted code patterns only.

**Comment by dstango** [comment]
You're right, if projects settings in eclipse are sufficient, there's no need for an extra plug-in.

My perspective was kind of clouded, because we have a mixture of IDEs (Eclipse, IntelliJ, VS Code) in them projects I mentioned.

This is of course quite unlikely for MoreUnit for Eclipse ;-) ... (Btw: what's the relationship to MoreUnit for IntelliJ? Do they share any code?)
